modules/hal_nordic/nrfx/CMakeLists.txt

```diff
@@ -38,6 +38,8 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF_SECURE_APPROTECT_LOCK
                                  ENABLE_SECURE_APPROTECT)
 zephyr_compile_definitions_ifdef(CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING
                                  ENABLE_SECURE_APPROTECT_USER_HANDLING)
+zephyr_compile_definitions_ifdef(CONFIG_NRF_TRACE_PORT
+                                 ENABLE_TRACE)
```

Comment:
why is this a `zephyr_compile_definitions_ifdef` and not a `zephyr_library_compile_definitions_ifdef` ?

Reply:
@anangl recommended doing the same as in 72e3bfe, and I wasn't aware of `zephyr_library_compile_definitions_ifdef`. This file use `zephyr_compile_definitions_ifdef` exclusively for definitions, but the library-alternative function for sources.

What are the practical differences between the two functions? I can change to use `zephyr_library_compile_definitions_ifdef` if that is the preferred function.